src/SprykerEco/Yves/Computop/Handler/PostPlace/ComputopEasyCreditPaymentHandler.php

```diff
@@ -40,30 +40,25 @@ public function handle(QuoteTransfer $quoteTransfer, array $responseArray)
      */
     protected function addPaymentToQuote(QuoteTransfer $quoteTransfer, AbstractTransfer $responseTransfer)
     {
-        if ($quoteTransfer->getPayment() !== null) {
-            $quoteTransfer->getPayment()->getComputopEasyCredit()->setEasyCreditInitResponse(
-                $responseTransfer
-            );
-
-            $quoteTransfer = $this->setPaymentProviderMethodSelection($quoteTransfer);
+        if ($quoteTransfer->getPayment()->getComputopEasyCredit() === null) {
+            $computopTransfer = new ComputopEasyCreditPaymentTransfer();
+            $quoteTransfer->getPayment()->setComputopEasyCredit($computopTransfer);
         }
 
+        $quoteTransfer->getPayment()->getComputopEasyCredit()->setEasyCreditInitResponse(
+            $responseTransfer
+        );
+
         return $quoteTransfer;
     }
 
     /**
      * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
      *
-     * @return \Generated\Shared\Transfer\QuoteTransfer
+     * @return void
```

Comment:
Please return value here as well.
In future TO's will be more correctly handled and we'll have to pass result value always.
let's get ready to this :)